lib/membrane_http_adaptive_stream/hls.ex

```diff
@@ -49,24 +46,80 @@ defmodule Membrane.HTTPAdaptiveStream.HLS do
     main_manifest_name = "#{manifest.name}.m3u8"
 
     case {tracks_by_content[:audio], tracks_by_content[:video]} do
-      {[audio], [video]} ->
+      {[audio], nil} ->
         [
-          {main_manifest_name, @av_manifest},
-          {"audio.m3u8", serialize_track(audio)},
-          {"video.m3u8", serialize_track(video)}
+          {main_manifest_name, build_master_playlist([audio], nil)},
+          {"audio.m3u8", serialize_track(audio)}
         ]
 
-      {[audio], nil} ->
-        [{main_manifest_name, serialize_track(audio)}]
+      {nil, videos} ->
+        List.flatten([
+          {main_manifest_name, build_master_playlist(nil, videos)},
+          videos
+          |> Enum.filter(&(&1.segments != @empty_segments))
+          |> Enum.map(&{build_media_playlist_path(&1), serialize_track(&1)})
+        ])
+
+      {[audio], videos} ->
+        List.flatten([
+          {main_manifest_name, build_master_playlist([audio], videos)},
+          {"audio.m3u8", serialize_track(audio)},
+          videos
+          |> Enum.filter(&(&1.segments != @empty_segments))
+          |> Enum.map(&{build_media_playlist_path(&1), serialize_track(&1)})
+        ])
 
-      {nil, [video]} ->
-        [{main_manifest_name, serialize_track(video)}]
+      {[_audio | _], _} ->
```

Comment:
I would move this to the top, just below `tracks_by_content = manifest.tracks |> Map.values() |> Enum.group_by(& &1.content_type)`

to create something like:
```
tracks_by_content = manifest.tracks |> Map.values() |> Enum.group_by(& &1.content_type)
if length(tracks_by_content[:audio]) > 1, do: raise ArgumentError, message: "Multiple audio tracks are not currently supported."
```